core/tools.js

```diff
@@ -1825,15 +1862,94 @@
 					}
 
 					function mapStyles( map ) {
-						ret.top = widths[ map[ 0 ] ];
-						ret.right = widths[ map[ 1 ] ];
-						ret.bottom = widths[ map[ 2 ] ];
-						ret.left = widths[ map[ 3 ] ];
+						ret.top = parts[ map[ 0 ] ];
+						ret.right = parts[ map[ 1 ] ];
+						ret.bottom = parts[ map[ 2 ] ];
+						ret.left = parts[ map[ 3 ] ];
 					}
 
 					return ret;
 				},
 
+				/**
+				 * Parses `border-style`, `border-width` and `border-color` shorthand styles into
+				 * border side shorthand styles.
+				 *
+				 * ```javascript
+				 * var styles = {
+				 *		'border-color': 'red blue',
+				 *		'border-style': 'solid dotted solid',
+				 *		'border-width': '1px 2px 3px 4px'
+				 * };
+				 *
+				 * console.log( CKEDITOR.tools.style.parse.splitBorderStyles( styles ) );
+				 * // Logs:
+				 * // {
+				 * // 	'border-top': '1px solid red',
+				 * // 	'border-right': '2px dotted blue',
+				 * // 	'border-bottom': '3px solid red',
+				 * // 	'border-left': '4px dotted blue'
+				 * // }
+				 *
+				 * // Use fallback to fill up missing style:
+				 * var missingColorStyles = {
+				 * 		'border-style': 'solid',
+				 * 		'border-width': '2px'
+				 * 	},
+				 * 	fallback = { color: 'red' };
+				 *
+				 * console.log( CKEDITOR.tools.style.parse.splitBorderStyles( missingColorStyles, fallback ) );
+				 * // Logs:
+				 * // {
+				 * // 	'border-top': '2px solid red',
+				 * // 	'border-right': '2px solid red',
+				 * // 	'border-bottom': '2px solid red',
+				 * // 	'border-left': '2px solid red'
+				 * // }
+				 * ```
+				 * @since 4.12.0
+				 * @param {Object} styles Border styles shorthand object.
+				 * @param {Object} [styles.border-color] Border color shorthand.
+				 * @param {Object} [styles.border-style] Border style shorthand.
+				 * @param {Object} [styles.border-width] Border width shorthand.
+				 * @param {Object} [fallback] Fallback object used to fill up missing style.
+				 * @param {Object} [fallback.color] Color CSS style used in absence of `border-color` style.
+				 * @param {Object} [fallback.style] Style CSS style used in absence of `border-style` style.
+				 * @param {Object} [fallback.width] Width CSS style used in absence of `border-width` style.
+				 * @returns {Object}
+				 * @returns {String} return.border-top Border top style.
+				 * @returns {String} return.border-right Border right style.
+				 * @returns {String} return.border-bottom Border bottom style.
+				 * @returns {String} return.border-left Border left style.
+				 * @member CKEDITOR.tools.style.parse
+				 */
+				splitBorderStyles: function( styles, fallback ) {
```

Comment:
Naming: it's already in style namespace, so there's no reason to call it `splitBorderStyles` because it would be `CKEDITOR.tools.style.parse.splitBorderStyles` (note style twice).

`CKEDITOR.tools.style.parse.splitBorder` is definitely simpler.